### PR TITLE
Update monitor name

### PR DESCRIPTION
**Pull request by FarajiA**
## JIRA:
See ticket here - https://jira.rax.io/browse/MNRVA-468

 ## Description:
Update monitor name - refactored updateMonitor service and accounted for the view subscription of the monitor details.

 ## Testing:
`npm run test`

 ## Screenshots:
updateMonitorName

### CODE REVIEW

**Comment by FarajiA** [comment]
When I run the tests I'm getting this warning:

`WARN: '
    It looks like you're using ngModel on the same form field as formControlName. 
    Support for using the ngModel input property and ngModelChange event with 
    reactive form directives has been deprecated in Angular v6 and will be removed 
    in Angular v7.
    
    For more information on this, see our API docs here:
    https://angular.io/api/forms/FormControlName#use-with-ngmodel
`

So I'll fix this as well before finalizing this PR, in the meantime what's here can be reviewed.

**Comment by lovedeep92** [comment]
> When I run the tests I'm getting this warning:
> 
> ```
> WARN: '
>     It looks like you're using ngModel on the same form field as formControlName. 
>     Support for using the ngModel input property and ngModelChange event with 
>     reactive form directives has been deprecated in Angular v6 and will be removed 
>     in Angular v7.
>     
>     For more information on this, see our API docs here:
>     https://angular.io/api/forms/FormControlName#use-with-ngmodel
> ```
> 
> So I'll fix this as well before finalizing this PR, in the meantime what's here can be reviewed.

File: src/app/_features/monitors/pages/details/monitor-details.page.html [hunk 1]
```diff
@@ -10,7 +10,31 @@
...
       <div class="hxCol hxSpan-9 monitor-title">
-        <h2>{{monitor.name}}</h2>
+        <h2 class="inline-icon">{{monitor.name}}</h2><hx-disclosure aria-controls="editMonPopover" #updateMonPen id="updateMonNamePen" class="editclick">
+          <hx-icon type="pencil"></hx-icon></hx-disclosure>
+        <hx-popover id="editMonPopover" position="bottom-right" class="popExpand">
+          <header>
+            <h5>Rename Monitor</h5>
+          </header>
+          <hx-div>
+          <form id="updateMonNameForm" [formGroup]="updateMonNameForm" (ngSubmit)="this.updateMonNameLoading = true; updateMonitorName(updateMonNameForm)">
```
Comment (lovedeep92) [inline src/app/_features/monitors/pages/details/monitor-details.page.html:20]: this.updateMonNameLoading is only value checking here. Do we need type checking as well, like === ?

File: src/app/_features/monitors/pages/details/monitor-details.page.ts [hunk 2]
```diff
@@ -10,8 +10,14 @@ import { DynamicFormComponent } from '../../components/dynamic-form/dynamic-form
...
 import { duration} from "moment";
 import { FieldConfig } from '../../interfaces/field.interface';
+import { FormGroup, FormBuilder } from '@angular/forms';
 
 declare const window: any;
+export enum UpdateSection {
+  additional = "additional",
+  name = "name",
+  plugin = "plugin"
+}
```
Comment (pras7430) [inline src/app/_features/monitors/pages/details/monitor-details.page.ts:20]: tight!

File: src/app/_features/monitors/pages/details/monitor-details.page.html (hunk 1 above)
Comment (FarajiA) [reply src/app/_features/monitors/pages/details/monitor-details.page.html:20]: Will update this submit and place the assignment in the update function.

File: src/app/_features/monitors/pages/details/monitor-details.page.html [hunk 3]
```diff
@@ -10,7 +10,31 @@
         <app-breadcrumb [routeDetails]="monitor.name || monitor.details.plugin.type"></app-breadcrumb>
       </div>
       <div class="hxCol hxSpan-9 monitor-title">
-        <h2>{{monitor.name}}</h2>
+        <h2 class="inline-icon">{{monitor.name}}</h2><hx-disclosure aria-controls="editMonPopover" #updateMonPen id="updateMonNamePen" class="editclick">
```
Comment (lovedeep92) [inline src/app/_features/monitors/pages/details/monitor-details.page.html:13]: I tried edit name of monitor and go back to list but the changes not replicated.

**Pull request merged**